From df97ea2570a82d73a1ed90736df341db3f0ea8de Mon Sep 17 00:00:00 2001
Message-Id: <df97ea2570a82d73a1ed90736df341db3f0ea8de.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Deepak M <m.deepak@intel.com>
Date: Sun, 25 Jan 2015 14:47:42 +0530
Subject: [PATCH 13/95] MUST_REBASE [VPG]: drm/i915: Add wrapper function to
 get display configuration

This function adds a wrapper function in I915 driver, to get
the display configurations from parsed VBT like no of devices,
no of child devices etc.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5193
Change-Id: I44f675655efc50333571bd32ea9230791dbc2dbe
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 drivers/gpu/drm/i915/i915_adf_wrapper.c |    7 +++++++
 include/drm/i915_adf.h                  |    2 +-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.c b/drivers/gpu/drm/i915/i915_adf_wrapper.c
index 613e39e..81c2bed 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.c
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.c
@@ -189,4 +189,11 @@ u16 intel_get_dsi_port_frm_vbt(void)
 }
 EXPORT_SYMBOL(intel_get_dsi_port_frm_vbt);
 
+void intel_get_vbt_disp_conf(void **child_dev, int *child_dev_num)
+{
+	*child_dev = (void *)i915_adf_dev->vbt.child_dev;
+	*child_dev_num = i915_adf_dev->vbt.child_dev_num;
+}
+EXPORT_SYMBOL(intel_get_vbt_disp_conf);
+
 #endif
diff --git a/include/drm/i915_adf.h b/include/drm/i915_adf.h
index fe8cf44..98aabc2 100644
--- a/include/drm/i915_adf.h
+++ b/include/drm/i915_adf.h
@@ -58,7 +58,7 @@ extern void intel_adf_display_rpm_put(void);
 extern int i915_adf_driver_initialize(void);
 extern u8 intel_adf_get_platform_id(void);
 extern u16 intel_get_dsi_port_frm_vbt(void);
-
+extern void intel_get_vbt_disp_conf(void **child_dev, int *child_dev_num);
 #endif
 
 
-- 
1.7.9.5

